client/app/account/components/DeleteAccount.tsx

```diff
@@ -1,45 +1,61 @@
+import React, { useState } from 'react'
 import { FiTrash } from 'react-icons/fi'
-import {
-  AlertDialog,
-  AlertDialogAction,
-  AlertDialogCancel,
-  AlertDialogContent,
-  AlertDialogDescription,
-  AlertDialogFooter,
-  AlertDialogHeader,
-  AlertDialogTitle,
-  AlertDialogTrigger,
-  Button,
-} from '@/_components/ui'
+import { Button } from '@/_components/ui'
+import { HouseType, UserType } from '@/_types'
 import { useDeleteAccount } from '../hooks'
+import { DeleteAccountModal, AdminWarningModal } from '.'
 
-const DeleteAccount = () => {
+type DeleteAccountProps = {
+  user: UserType
+  userHouse: HouseType | undefined
+}
+
+const DeleteAccount = ({ user, userHouse }: DeleteAccountProps) => {
+  const [isOpenDeleteModal, setIsOpenDeleteModal] = useState(false)
+  const [isOpenAdminWarningModal, setIsOpenAdminWarningModal] = useState(false)
   const { onDelete, isPending } = useDeleteAccount()
 
+  const handleDeleteAccount = () => {
+    if (!userHouse) {
+      setIsOpenDeleteModal(true)
+      return
+    }
+
+    const isAdmin = userHouse.houseMembers.filter((member) => member.isAdmin)
+
+    if (isAdmin.length === 1 && isAdmin[0].id === user.id) {
+      setIsOpenAdminWarningModal(true)
```

Comment:
Currently, I can't delete my account when I'm the only house member, but the user should be able to delete their account and the house should be deleted at the same time, right?

Reply:
Oh you are assuming that the house should be deleted first

Reply:
The user can delete the account when the user is the sole house member (& sole admin) after the user deletes the house. That's why I put the warning message ` Before proceeding, please assign another admin or delete the house in the House Settings.`. I added `or delete the house in the House Settings.`

Reply:
Okay! I thought it's more user-friendly that when the user deleted their account and is the only house member, the house will be automatically deleted, but this is also fine too, so you can keep this way!